akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -439,6 +442,7 @@ private[akka] class ShardRegion(
     case query: ShardRegionQuery                 ⇒ receiveQuery(query)
     case msg: RestartShard                       ⇒ deliverMessage(msg, sender())
     case msg if extractEntityId.isDefinedAt(msg) ⇒ deliverMessage(msg, sender())
```

Comment:
In case they (accidentally) handle `StartEntity` also in `extractEntityId` we could be nice and call `deliverStartEntity` from `deliverMessage`. For example the `HashCodeMessageExtractor` does that, and the Java API `MessageExtractor` is an interface for both.